pkg/api/universal/conversation_stream.go

```diff
@@ -0,0 +1,312 @@
+	// Try to get the underlying LangChain Go model for streaming support
+	if streamer, ok := component.(conversation.StreamingConversation); ok {
+		return p.processRealStreaming(ctx, stream, streamer, request)
+	}
+
+	// Fallback: simulate streaming with existing Converse method
+	return p.processSimulatedStreaming(ctx, stream, component, request)
+}
+
+// processRealStreaming handles true LangChain Go streaming
+func (p *StreamingPipelineImpl) processRealStreaming(
+	ctx context.Context,
+	stream runtimev1pb.Dapr_ConverseStreamAlpha1Server,
+	streamer conversation.StreamingConversation,
+	request *conversation.ConversationRequest,
+) (*runtimev1pb.ConversationUsage, error) {
+	var finalResp *conversation.ConversationResponse
+	var streamErr error
+
+	// Create the streaming function that LangChain Go will call for each chunk
+	streamFunc := func(chunkCtx context.Context, chunk []byte) error {
+		// Process chunk through middleware chain
+		processedChunk := p.processChunkThroughMiddleware(chunk)
+
+		// Send processed chunk immediately (if any ready)
+		if len(processedChunk) > 0 {
+			if err := p.sendChunk(stream, processedChunk); err != nil {
+				p.logger.Errorf("Failed to send streaming chunk: %v", err)
+				return err
+			}
+		}
+
+		return nil
+	}
+
+	// Call the component's streaming method
+	finalResp, streamErr = streamer.ConverseStream(ctx, request, streamFunc)
+
+	// Flush any remaining content in middleware buffers
+	if remaining := p.flushMiddleware(); len(remaining) > 0 {
+		if sendErr := p.sendChunk(stream, remaining); sendErr != nil {
+			p.logger.Errorf("Failed to send final streaming chunk: %v", sendErr)
+			// Return the component error if it exists, otherwise the send error
+			if streamErr != nil {
+				return nil, streamErr
+			}
+			return nil, sendErr
+		}
+	}
+
+	if streamErr != nil {
+		return nil, streamErr
+	}
+
+	// Send completion with context and usage stats
+	contextID := ""
+	var usage *runtimev1pb.ConversationUsage
+	if finalResp != nil {
+		contextID = finalResp.ConversationContext
+		usage = convertUsageToProto(finalResp)
+	}
+	return usage, p.sendComplete(stream, contextID, usage)
+}
+
+// processSimulatedStreaming falls back to chunking the complete response
+func (p *StreamingPipelineImpl) processSimulatedStreaming(
+	ctx context.Context,
+	stream runtimev1pb.Dapr_ConverseStreamAlpha1Server,
+	component conversation.Conversation,
+	request *conversation.ConversationRequest,
+) (*runtimev1pb.ConversationUsage, error) {
+	// Call the regular Converse method
+	resp, err := component.Converse(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	var contextID string
+
+	// Simulate streaming by sending the complete response as chunks
+	if resp != nil {
+		contextID = resp.ConversationContext
+		if len(resp.Outputs) > 0 {
+			for _, output := range resp.Outputs {
+				// Break the result into chunks to simulate streaming
+				content := output.Result
+				chunkSize := 50 // Send in 50-character chunks
```

Comment:
Why `50`?

gRPC by default has a maximum message size of 4MB (~4,000,000 bytes)- It's incredibly inefficient to be sending small payloads like this.

I don't think we need to be chunking at all. But if we must..  we should be fetching what the max message size configured is, and chunking on a slightly smaller number to account for headers and other fields in the message.